Spatial_Engine/Query/IsOnGrid.cs

```diff
@@ -0,0 +1,103 @@
+            ICurve gridCurve = grid.Curve.IProject(Plane.XY);
+
+            return position.IDistance(gridCurve) <= maxDistance;
+        }
+
+        /******************************************/
+
+        [Description("Checks if the geometrical representation of an IElement1D, projected onto the XY Plane, is within a set distance from a grid line.")]
+        [Input("element1D", "The IElement1D that will be checked whether it is on Grid.")]
+        [Input("grid", "The Grid to use for evaulation.")]
+        [Input("maxDistance", "The maximum distance allowed from the Grid for this method to return true.", typeof(Length))]
+        [Output("isOnGrid", "A boolean which is true if the geometrical representation of an IElement1D is within a set distance from the grid line.")]
+        public static bool IsOnGrid(this IElement1D element1D, Grid grid, double maxDistance)
+        {
+            ICurve curve = element1D.IGeometry().IProject(Plane.XY);
+            ICurve gridCurve = grid.Curve.IProject(Plane.XY);
+
+            return curve.Distance(gridCurve) <= maxDistance;
+        }
+
+        /******************************************/
+
+        [Description("Checks if the geometrical representation of an IElement2D, projected onto the XY Plane, is within a set distance from a grid line.")]
+        [Input("element2D", "The IElement2D that will be checked whether it is on Grid.")]
+        [Input("grid", "The Grid to use for evaulation.")]
+        [Input("maxDistance", "The maximum distance allowed from the Grid for this method to return true.", typeof(Length))]
+        [Output("isOnGrid", "A boolean which is true if the geometrical representation of an IElement2D is within a set distance from the grid line.")]
+        public static bool IsOnGrid(this IElement2D element2D, Grid grid, double maxDistance)
+        {
+            List<IElement1D> elements1D = element2D.IOutlineElements1D();
+
+            foreach (IElement1D e1D in elements1D)
+                if (IsOnGrid(e1D, grid, maxDistance))
+                    return true;
+
+            return false;
+        }
+
+        /******************************************/
+        /****   Public Methods - Interfaces    ****/
+        /******************************************/
+
+        [Description("Checks if the geometrical representation of an IElement, projected onto the XY Plane, is within a set distance from a grid line.")]
+        [Input("element", "The IElement that will be checked whether it is on Grid.")]
+        [Input("grid", "The Grid to use for evaulation.")]
+        [Input("maxDistance", "The maximum distance allowed from the Grid for this method to return true.", typeof(Length))]
+        [Output("isOnGrid", "A boolean which is true if the geometrical representation of the IElement is within a set distance from the grid line.")]
+        public static bool IsOnGrid(this IElement element, Grid grid, double maxDistance)
```

Comment:
Ah, that makes a lot of sense! I added that method simply because it seemed like most other methods had one, while not asking anybody first and not being aware of the norms beyond that.